dspace-server-webapp/src/test/java/org/dspace/app/rest/CollectionGroupRestControllerIT.java

```diff
@@ -0,0 +1,2094 @@
+        metadataRest.put("dc.subject", new MetadataValueRest("testSubject"));
+
+        groupRest.setMetadata(metadataRest);
+
+        authorizeService.addPolicy(context, parentCommunity, Constants.ADMIN, eperson);
+
+        AtomicReference<UUID> idRef = new AtomicReference<>();
+
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(post("/api/core/collections/" + collection.getID() + "/submittersGroup")
+                                     .content(mapper.writeValueAsBytes(groupRest))
+                                     .contentType(contentType))
+                        .andExpect(status().isCreated())
+                        .andDo(result -> idRef
+                            .set(UUID.fromString(read(result.getResponse().getContentAsString(), "$.id")))
+                        );
+        Group submittersGroup = groupService.find(context, idRef.get());
+        getClient(token).perform(get("/api/core/collections/" + collection.getID() + "/submittersGroup"))
+                        .andExpect(status().isOk())
+                        .andExpect(
+                            jsonPath("$",
+                                     GroupMatcher.matchGroupEntry(submittersGroup.getID(), submittersGroup.getName())));
+
+    }
+
+    @Test
+    public void postCollectionSubmittersGroupCreateSubmittersGroupSuccessCollectionAdmin() throws Exception {
+
+        ObjectMapper mapper = new ObjectMapper();
+        GroupRest groupRest = new GroupRest();
+        MetadataRest metadataRest = new MetadataRest();
+        metadataRest.put("dc.description", new MetadataValueRest("testingDescription"));
+        metadataRest.put("dc.subject", new MetadataValueRest("testSubject"));
+
+        groupRest.setMetadata(metadataRest);
+
+        authorizeService.addPolicy(context, collection, Constants.ADMIN, eperson);
+
+        AtomicReference<UUID> idRef = new AtomicReference<>();
+
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(post("/api/core/collections/" + collection.getID() + "/submittersGroup")
+                                     .content(mapper.writeValueAsBytes(groupRest))
+                                     .contentType(contentType))
+                        .andExpect(status().isCreated())
+                        .andDo(result -> idRef
+                            .set(UUID.fromString(read(result.getResponse().getContentAsString(), "$.id")))
+                        );
+        Group submittersGroup = groupService.find(context, idRef.get());
+        getClient(token).perform(get("/api/core/collections/" + collection.getID() + "/submittersGroup"))
+                        .andExpect(status().isOk())
+                        .andExpect(
+                            jsonPath("$",
+                                     GroupMatcher.matchGroupEntry(submittersGroup.getID(), submittersGroup.getName())));
+
```

Comment:
this group must be deleted in a finally block

Reply:
This group will be deleted by the builders as this group is linked to the collection object.